src/sync/Synchroniser.js

```diff
@@ -244,7 +252,11 @@ export class Synchroniser {
       throw new Error('Unexpected response from sync server');
     }
     if (responseJson.error.length > 0) {
-      throw new Error('Server rejected pushed records');
+      if (responseJson.error.startsWith('Site registration doesn\'t match.')) {
+        throw new Error(responseJson.error);
+      } else {
+        throw new Error('Server rejected pushed records');
```

Comment:
We don't seem to inform of error here. I would suggest instead of 'Server rejected pushed record' -> `Server rejected pushed record (${responseJson.error)` <- would work for both cases + extra information in case there is error.

Reply:
@andreievg this related #680

Reply:
Thanks @andreievg I've updated comments there about to handle the codes for these both issues.